### ISSUE TITLE
Version11x vendored build output for windows-gnu target has wrong file name

### ISSUE THREAD

**Issue by zonyitoo** [sfackler/rust-openssl#1256]
When cross compiling openssl with `vendored` feature enabled, 

```rust
    let libs = match libs_env.as_ref().and_then(|s| s.to_str()) {
        Some(ref v) => v.split(":").collect(),
        None => match version {
            Version::Openssl10x if target.contains("windows") => vec!["ssleay32", "libeay32"],
            Version::Openssl11x if target.contains("windows") => vec!["libssl", "libcrypto"],
            _ => vec!["ssl", "crypto"],
        },
    };
```

`determine_mode` is trying to find `liblibcrypto.a` and `liblibssl.a`:

```rust
    let can_static = libs
        .iter()
        .all(|l| files.contains(&format!("lib{}.a", l)) || files.contains(&format!("{}.lib", l)));
    let can_dylib = libs.iter().all(|l| {
        files.contains(&format!("lib{}.so", l))
            || files.contains(&format!("{}.dll", l))
            || files.contains(&format!("lib{}.dylib", l))
    });
```

because `libs` are `vec!["libssl", "libcrypto"]`. But that's incorrect, it should be `vec!["ssl", "crypto"]`:

```
$ ls ./target/x86_64-pc-windows-gnu/release/build/openssl-sys-fa4c162fa3e2358b/out/openssl-build/install/lib
libcrypto.a libssl.a    pkgconfig
```

A simple fix is to set `OPENSSL_LIBS` with

```bash
export OPENSSL_LIBS="ssl:crypto"
```

But it would be better if `main.rs` could fix this issue.

**Comment by laumann** [comment]
Does #1261 (or #1044) fix this? It's a neat little workaround though...

**Comment by zonyitoo** [comment]
Both of them use similiar approach. I think they will work.

**Comment by laumann** [comment]
@zonyitoo The former is just a rebase on master of the latter :-)

**Comment by laumann** [comment]
Obviously I forgot to add a `Fixes` line to the commit, but #1265 should've fixed this issue.

**Comment by zonyitoo** [comment]
Great!
